airflow/jobs/scheduler_job_runner.py

```diff
@@ -304,26 +323,24 @@ def _executable_task_instances_to_queued(self, max_tis: int, session: Session) -
         starved_pools = {pool_name for pool_name, stats in pools.items() if stats["open"] <= 0}
 
         # dag_id to # of running tasks and (dag_id, task_id) to # of running tasks.
-        dag_active_tasks_map: DefaultDict[str, int]
-        task_concurrency_map: DefaultDict[tuple[str, str], int]
-        dag_active_tasks_map, task_concurrency_map = self.__get_concurrency_maps(
-            states=list(EXECUTION_STATES), session=session
-        )
+        concurrency_map = self.__get_concurrency_maps(states=list(EXECUTION_STATES), session=session)
```

Comment:
Since `states` is only used in `in_()` (which takes any iterable), I think the `list()` conversion is also unnecessary.

Reply:
I changed the argument type from `list[TaskInstanceState]` to `Iterable[TaskInstanceState]` and removed the conversion